app/generateSMTScriptProp.ts

```diff
@@ -0,0 +1,7 @@
+interface Smt { smt2: string; propositions: Set<string>; }
+
+export default function generateSMTScriptProp(userSmt: Smt, sysSmt: Smt): string {
+  const declarationsArray = Array.from(new Set([...userSmt.propositions, ...sysSmt.propositions])).map((d) => `(declare-const ${d} Bool)\n`);
+  declarationsArray.push(`(assert (not (= ${userSmt.smt2} ${sysSmt.smt2})))`);
+  return declarationsArray.join("");
```

Comment:
When composing lines for things, you want to keep things in arrays for as long as possible, stringifying at the last minute. That means that debugging / refactoring becomes much easier. If you want to add something somewhere, you can go to the correct spot in the array instead of pulling apart the string again.